lib/install/helpers.rb

```diff
@@ -10,7 +10,9 @@ def bundler_run_cmd
   end
 
   def using_bun?
-    File.exist?('bun.lockb') || (tool_exists?('bun') && !File.exist?('yarn.lock'))
+    tool_exists?('bun') && (File.exist?('bun.lockb') || 
+                            File.exist?('bun.lock') ||       
+                            File.exist?('yarn.lock'))
```

Comment:
When you have bun installed on the system, but you are building the project with yarn and having a yarn.lock, i think we should remove the File.exist?('yarn.lock') even if bun manage yarn.lock, to avoid the confusion.